File: app/src/main/java/com/pixels/blockies/app/game/HighScore.java
package com.pixels.blockies.app.game;

import android.content.SharedPreferences;

/**
 * The High-Score that is persisted in the SharedPreferences
 */
public class HighScore {

    private static final String KEY = "highscore";

    SharedPreferences dataStore = null;

    public HighScore(SharedPreferences dataStore){
        this.dataStore = dataStore;
    }

    public int getScore(){
        return dataStore.getInt(KEY, 0);
    }

    public void saveScore(int newScore){
        int oldScore = dataStore.getInt(KEY, 0);
        if(newScore > oldScore ){
            SharedPreferences.Editor edit = dataStore.edit();
            edit.putInt(KEY, newScore);
            edit.commit();
        }

    }

}

File: app/src/main/java/com/pixels/blockies/game/game/HighScore.kt
package com.pixels.blockies.game.game

import android.content.SharedPreferences

/**
 * The High-Score that is persisted in the SharedPreferences
 */
class HighScore(dataStore: SharedPreferences?) {
    var dataStore: SharedPreferences? = null

    init {
        this.dataStore = dataStore
    }

    fun getScore(): Int {
        return dataStore!!.getInt(KEY, 0)
    }

    fun saveScore(newScore: Int) {
        val oldScore = dataStore!!.getInt(KEY, 0)
        if (newScore > oldScore) {
            val edit = dataStore!!.edit()
            edit.putInt(KEY, newScore)
            edit.commit()
        }
    }

    companion object {
        private const val KEY = "highscore"
    }
}